src/lib.rs

```diff
@@ -119,23 +124,28 @@ impl fmt::Display for Cid {
     }
 }
 
+impl FromStr for Cid {
+    type Err = Error;
+    fn from_str(src: &str) -> Result<Self> {
+        src.to_cid()
+    }
+}
+
 impl Prefix {
     pub fn new_from_bytes(data: &[u8]) -> Result<Prefix> {
-        let mut cur = Cursor::new(data);
-
-        let raw_version = cur.read_varint()?;
-        let raw_codec = cur.read_varint()?;
-        let raw_mh_type: u64 = cur.read_varint()?;
-
+        let (raw_version, remain) = varint_decode::u64(data)?;
         let version = Version::from(raw_version)?;
+
+        let (raw_codec, remain) = varint_decode::u64(remain)?;
         let codec = Codec::from(raw_codec)?;
 
+        let (raw_mh_type, remain) = varint_decode::u64(remain)?;
         let mh_type = match multihash::Code::from_u64(raw_mh_type) {
-            multihash::Code::Custom(_) => Err(Error::UnknownCodec),
-            code => Ok(code),
-        }?;
+            multihash::Code::Custom(_) => return Err(Error::UnknownCodec),
+            code => code,
+        };
 
-        let mh_len = cur.read_varint()?;
+        let mh_len = varint_decode::usize(remain)?.0;
```

Comment:
No need to change it, I just bring it up as I like to use code review also to learn from others. I would write this on as `let (mh_len, _remain) = varint_decode::usize(remain)?;`. That is more consistent with other calls and I think it makes this easier to read/understand what is going on (that you ignore the remaining bytes).

Reply:
I think your suggestion makes sense